### ISSUE TITLE
Extend to Multi-class Classification Task and Be compatible with PyTorch scheduler

### ISSUE THREAD

**Issue by Cogito2012** [yzhuoning/LibAUC#1]
Hi Zhuoning,

This is an interesting work! 
I am wondering if the DAM method can be extended to a multi-class classification task with long-tailed imbalanced data. Intuitively, this should be possible as the famous [sklearn tool ](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score) provides auc score for multi-class setting by using one-versus-rest or one-versus-one technique.

Besides, it seems that `optimizer.update_regularizer()` is called only when the learning rate is reduced, thus it would be more elegant to incorporate this functional call into a pytorch lr scheduler. E.g., 

```python
scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer)
scheduler.step()    # override the step to fulfill: optimizer.update_regularizer()

```
For current libauc version, the PESG optimizer is not compatible with schedulers in `torch.optim.lr_scheduler` . It would be great if this feature can be supported in the future.

Thanks for your work!

**Comment by yzhuoning** [comment]
Hi Wentao, 

Thanks for your suggestions!

In our new implementations, `PESG` now can work with different Pytorch learning schedulers, and please see [example](https://github.com/yzhuoning/LibAUC/blob/main/examples/03_Training_with_Pytorch_Learning_Rate_Scheduling.ipynb) here. However, you still need to manually do `optimizer.update_regularizer()` to reset the stage.  We will eventually combine these two functions. For multi-class extension, it is indeed possible. I have done some experiments on the long-tailed version of CIRFAR datasets and I will update more details in the future updates. 

Thanks!

**Comment by Cogito2012** [comment]
Thanks for the update! I'll take a look at the example.
